shiny/ui/datagrid/js/tsconfig.json

```diff
@@ -1,14 +1,15 @@
 {
   "compilerOptions": {
-    "module": "commonjs",
-    "lib": ["es5", "es6", "dom"],
+    "module": "ES2020",
+    "lib": ["ES2021", "DOM"],
```

Comment:
What are the browser compatibility implications of this?

Reply:
If you go to https://caniuse.com/?search=es2021 and click on Feature List, then scroll all the way to the bottom and click on Date Relative, you'll see this:
<img width="1057" alt="image" src="https://github.com/rstudio/py-shiny/assets/86978/6595734c-4fd4-4b31-becc-511b56ffd1f7">


The short story is that all major browsers have had support for all these features since early 2021.

I tried the same search for es2022, and found that Safari only got full support for es2022 features this year, which is why I went with es2021.
<img width="1093" alt="image" src="https://github.com/rstudio/py-shiny/assets/86978/2b8384ce-29e4-48e6-bfcc-0d6443bae306">

Reply:
Whoa, never saw that view, that's super cool. Though of course it's red/green 😠